elide-core/src/main/java/com/yahoo/elide/graphql/operations/UpdateOperation.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2017, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.graphql.operations;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.HttpStatus;
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.graphql.Environment;
+import com.yahoo.elide.graphql.PersistentResourceFetcher;
+import graphql.language.Field;
+import graphql.language.ObjectValue;
+import graphql.language.StringValue;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLType;
+
+import javax.ws.rs.WebApplicationException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.yahoo.elide.graphql.ModelBuilder.ARGUMENT_DATA;
+import static com.yahoo.elide.graphql.ModelBuilder.ARGUMENT_ID;
+
+/**
+ * Encapsulate the GraphQL update logic.
+ */
+public class UpdateOperation {
+    private final Field field;
+    private final RequestScope requestScope;
+    private final EntityDictionary dictionary;
+    private final String id;
+    private final Object source;
+    private final GraphQLType outputType;
+    private final List<Map<String, Object>> requestData;
+    private final boolean isRoot;
+    private final Class entityClass;
+    private final PersistentResource resource;
+    private final String idFieldName;
+
+    public UpdateOperation(Environment environment) {
+        field = environment.field;
+        requestScope = environment.requestScope;
+        dictionary = requestScope.getDictionary();
+        id = environment.id.orElse(null);
+        source = environment.source;
+        outputType = environment.outputType;
+        requestData = environment.data;
+        isRoot = !(source instanceof PersistentResource);
+        resource = (isRoot) ? null : (PersistentResource) source;
+        entityClass = (isRoot)
+                ? dictionary.getEntityClass(field.getName())
+                : dictionary.getParameterizedType(((PersistentResource) source).getObject(), field.getName());
+        idFieldName = dictionary.getIdFieldName(entityClass);
+    }
+
+    public Object execute() {
+        Collection<PersistentResource> objectsToUpdate = fetchResourceCollection();
+
+        if (outputType instanceof GraphQLList) {
```

Comment:
The outputType could be a single GraphQLOutputObject.  This assumption is not correct.